Core/Src/main.c

```diff
@@ -59,106 +81,133 @@ void SystemClock_Config(void);
 /* USER CODE END 0 */
 
 /**
-  * @brief  The application entry point.
-  * @retval int
-  */
-int main(void)
-{
-  /* USER CODE BEGIN 1 */
+ * @brief  The application entry point.
+ * @retval int
+ */
+int main(void) {
+    /* USER CODE BEGIN 1 */
+
+    /* USER CODE END 1 */
+
+    /* MCU Configuration--------------------------------------------------------*/
+
+    /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+    HAL_Init();
+
+    /* USER CODE BEGIN Init */
+
+    /* USER CODE END Init */
 
-  /* USER CODE END 1 */
+    /* Configure the system clock */
+    SystemClock_Config();
 
-  /* MCU Configuration--------------------------------------------------------*/
+    /* USER CODE BEGIN SysInit */
 
-  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
-  HAL_Init();
+    /* USER CODE END SysInit */
 
-  /* USER CODE BEGIN Init */
+    /* Initialize all configured peripherals */
+    MX_GPIO_Init();
+    MX_USART2_UART_Init();
+    MX_SPI1_Init();
+    MX_TIM1_Init();
+    /* USER CODE BEGIN 2 */
 
-  /* USER CODE END Init */
+    // Setting CS high to signal no SPI packets
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
 
-  /* Configure the system clock */
-  SystemClock_Config();
+    // Setting duty cycle and starting PWM
+    __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, MIN_PULSE_WIDTH);
+    HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
 
-  /* USER CODE BEGIN SysInit */
+    /* USER CODE END 2 */
 
-  /* USER CODE END SysInit */
+    /* Infinite loop */
+    /* USER CODE BEGIN WHILE */
+    while (1) {
+        // Setting CS low to signal start of SPI packet
+        HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
 
-  /* Initialize all configured peripherals */
-  MX_GPIO_Init();
-  MX_USART2_UART_Init();
-  /* USER CODE BEGIN 2 */
+        // I should probably check the return value here
+        HAL_SPI_TransmitReceive(&hspi1, (uint8_t*) &ADC_TX_BYTES,
+                (uint8_t*) &adc_out, 2, SPI_COMM_TIMEOUT);
```

Comment:
90% correct here. The reason we are using this `HAL_SPI_TransmitReceive()` instead of `HAL_SPI_Transimit` `HAL_SPI_Receive` separately is because the TX and RX in this scenario are happening at the same time. As the datasheet suggests, the first and half byte matters for TX, and devices don't care the first one and half byte of data in RX, but the last one and half RX is your useful data. So yo should create array size of 3.